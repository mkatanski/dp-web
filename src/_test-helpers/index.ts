export * from "./withResponsiveContext";
export * from "./withThemeProvider";
