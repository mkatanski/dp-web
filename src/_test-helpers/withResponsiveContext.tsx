import React from "react";
import { Context as ResponsiveContext } from "react-responsive";
import { Breakpoints } from "config/breakpoints";

export const withMobileSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.mobile }}>
    {Component}
  </ResponsiveContext.Provider>
);

export const withTabletSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.tablet }}>
    {Component}
  </ResponsiveContext.Provider>
);

export const withDesktopSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.desktop }}>
    {Component}
  </ResponsiveContext.Provider>
);

export const withLargeDesktopSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.largeDesktop }}>
    {Component}
  </ResponsiveContext.Provider>
);
