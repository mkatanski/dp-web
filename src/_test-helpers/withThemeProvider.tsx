import React from "react";
import { ThemeProvider as DefaultThemeProvider } from "lib/theme/ThemeProvider";
import { StyledTheme, Theme } from "styles/theme/_types";
import { createMuiTheme } from "@material-ui/core";

const testTheme: StyledTheme = {
  primary: "primary",
  secondary: "secondary",
  textPrimary: "textPrimary",
  textSecondary: "textSecondary",
  textInverse: "textInverse"
};

const muiTheme = createMuiTheme({
  direction: "ltr"
});

const theme: Theme = Object.assign(testTheme, muiTheme);

export const withThemeProvider = (Component: React.ReactNode) => (
  <DefaultThemeProvider theme={theme}>{Component}</DefaultThemeProvider>
);
