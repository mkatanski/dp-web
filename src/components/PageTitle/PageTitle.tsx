import React from "react";
import { Grid as BareGrid, Typography } from "@material-ui/core";
import styled from "styled-components";

export type PageTitleProps = {
  category: string;
  title: string;
};

const Grid = styled(BareGrid)`
  margin-bottom: ${({ theme }) => `${theme.spacing(2)}px`};
`;

export const PageTitle: React.FC<PageTitleProps> = ({
  category,
  title
}: PageTitleProps) => {
  return (
    <Grid alignItems="flex-end" container justify="space-between" spacing={2}>
      <BareGrid item>
        <Typography component="h2" variant="overline" color="textSecondary">
          {category}
        </Typography>
        <Typography component="h1" variant="h3" color="textPrimary">
          {title}
        </Typography>
      </BareGrid>
    </Grid>
  );
};
