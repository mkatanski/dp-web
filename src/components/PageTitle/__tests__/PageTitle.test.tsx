import React from "react";
import TestRenderer from "react-test-renderer";
import "jest-styled-components";
import { withThemeProvider } from "_test-helpers";

/* Uncomment below if needed */
// import { withThemeProvider, withDesktopSize } from 'testsHelpers';
// import { render, getByTestId } from "@testing-library/react";

import { PageTitle } from "../PageTitle";

/* Uncomment below if You need/want to use enzyme */
// import enzyme from "enzyme";

describe("components", () => {
  describe("PageTitle", () => {
    it("should render PageTitle component", () => {
      const tree = TestRenderer.create(
        withThemeProvider(<PageTitle category="category" title="title" />)
      );

      expect(tree).toMatchSnapshot();
    });
  });
});
