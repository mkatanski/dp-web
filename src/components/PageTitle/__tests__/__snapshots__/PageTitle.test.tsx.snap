// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components PageTitle should render PageTitle component 1`] = `
.c0 {
  margin-bottom: 16px;
}

<div
  className="c0"
  data-props={
    Object {
      "alignItems": "flex-end",
      "container": true,
      "justify": "space-between",
      "spacing": 2,
    }
  }
  id="MUI.Grid"
>
  <div
    data-props={
      Object {
        "item": true,
      }
    }
    id="MUI.Grid"
  >
    <div
      data-props={
        Object {
          "color": "textSecondary",
          "component": "h2",
          "variant": "overline",
        }
      }
      id="MUI.Typography"
    >
      category
    </div>
    <div
      data-props={
        Object {
          "color": "textPrimary",
          "component": "h1",
          "variant": "h3",
        }
      }
      id="MUI.Typography"
    >
      title
    </div>
  </div>
</div>
`;
